### PR TITLE
cilium-dbg: Replace statedb command with "shell -- db show"

### PR DESCRIPTION
**Pull request by joamaki**
```
cilium-dbg: Replace the statedb command with "shell db" commands

    The "cilium-dbg statedb" command now returns a deprecated text that shows
    how to use the "cilium-dbg shell -- db" commands. We'll remove it fully in
    next release. The "cilium-dbg statedb dump" will probably need to be kept as
    a hidden command for bugtool compatibility.

    This removes the need to list all tables we want to inspect in statedb.go
    and the need to import bunch of agent packages to pull in the object types.

    Signed-off-by: Jussi Maki <jussi@isovalent.com>

daemon: Add FromString to StateDB table indexes

    To support queries using a string with the "db" script commands, add
    "FromString" implementation to all StateDB indexes.

    Signed-off-by: Jussi Maki <jussi@isovalent.com>
```

tl;dr: `cilium-dbg statedb <table>` commands now becomes `cilium-dbg shell -- db show <table>`. `cilium-dbg statedb` shows a deprecated disclaimer and example usage of `shell -- db`.

### CODE REVIEW

**Comment by joamaki** [comment]
Hmm, since this is completely changing how "cilium-dbg statedb" command works, we could also just change the command to "cilium-dbg db". This way it'll also match the script command, and the help matches as well.

File: cilium-dbg/cmd/shell.go [hunk 1]
```diff
@@ -64,6 +64,7 @@ func dialShell(w io.Writer) (net.Conn, error) {
 }
 
 func shellExchange(w io.Writer, format string, args ...any) error {
+
```
Comment (tklauser) [inline cilium-dbg/cmd/shell.go]: stray newline

File: cilium-dbg/cmd/shell.go [hunk 2]
```diff
@@ -84,10 +85,7 @@ func shellExchange(w io.Writer, format string, args ...any) error {
 			return nil
 		case "[stdout]":
 		default:
-			if _, err := w.Write(line); err != nil {
-				return err
-			}
-			if _, err := w.Write(newline); err != nil {
+			if _, err := fmt.Fprintf(w, "%s\n", line); err != nil {
```
Comment (tklauser) [inline cilium-dbg/cmd/shell.go]: Could be simplified to:

```suggestion
			if _, err := fmt.Fprintln(w, line); err != nil {
```

**Review by tklauser — APPROVED** [review]
Looks good, nice and clean! Two nits inline.

I've left my comments wrt. killing the `statedb` subcommand in a [separate reply](https://github.com/cilium/cilium/pull/35545#issuecomment-2446933813) and I'll let you decide on how you want to move forward with this.

**Review by learnitall — APPROVED** [review]
Vendor changes LGTM :rocket:

**Comment by joamaki** [comment]
> Looks good, nice and clean! Two nits inline.
> 
> I've left my comments wrt. killing the `statedb` subcommand in a [separate reply](https://github.com/cilium/cilium/pull/35545#issuecomment-2446933813) and I'll let you decide on how you want to move forward with this.

Thanks! I think simplifying is the way to go, but I don't want to leave folks used to "cilium-dbg statedb" confused. I'll make the "cilium-dbg statedb" just spit out a "deprecated message" that tells how to do the same things with "cilium-dbg shell -- db show <table>" etc. so we have a smooth transition. And then kill the command in v1.17.

**Review by qmonnet — APPROVED** [review]

**Review by dylandreimerink — APPROVED** [review]

**Review by pippolo84 — APPROVED** [review]
egress-gateway changes LGTM :100:

**Review by nathanjsweet — APPROVED** [review]
